torch/nn/init.py

```diff
@@ -55,14 +97,69 @@ def constant(tensor, val):
     if isinstance(tensor, Variable):
         constant(tensor.data, val)
         return tensor
+
     return tensor.fill_(val)
 
 
+def eye(tensor):
+    """Fills the 2-dimensional input Tensor or Variable with the identity matrix. Preserves the identity of the inputs in
+    Linear layers.
+
+    Args:
+        tensor: a 2-dimensional torch.Tensor or autograd.Variable
+
+    Examples:
+        >>> w = torch.Tensor(3, 5)
+        >>> nn.init.eye(w)
+    """
+    if tensor.ndimension() != 2:
+        raise ValueError("Only tensors with 2 dimensions are supported")
+
+    if isinstance(tensor, Variable):
+        eye(tensor.data)
+        return tensor
+
+    return tensor.copy_(torch.eye(tensor.size(0), tensor.size(1)))
+
+
+def dirac(tensor, scaled=True):
+    """Fills the {3, 4, 5}-dimensional input Tensor or Variable with the Dirac delta function. Preserves the identity of
+    the inputs in Convolutional layers.
+
+    Args:
+        tensor: a {3, 4, 5}-dimensional torch.Tensor or autograd.Variable
+        scaled: a boolean indicating that the output of the sum of the filters will be the identity
+
+    Examples:
+        >>> w = torch.Tensor(3, 16, 5, 5)
+        >>> nn.init.dirac(w)
+    """
+    dimensions = tensor.ndimension()
+    if dimensions not in [3, 4, 5]:
+        raise ValueError("Only tensors with 3, 4, or 5 dimensions are supported")
+
+    if isinstance(tensor, Variable):
+        dirac(tensor.data, scaled=scaled)
+        return tensor
+
+    tensor.zero_()
+    val = 1.0 / tensor.size(1) if scaled else 1
+
+    if dimensions == 3:  # Temporal convolution
+        tensor[:, :, tensor.size(2) // 2].fill_(val)
+    elif dimensions == 4:  # Spatial convolution
+        tensor[:, :, tensor.size(2) // 2, tensor.size(3) // 2].fill_(val)
```

Comment:
I'm a bit confused about the definition of `dirac` in here. This filter is actually performing the sum over the input channels, so it doesn't give you the identity, but a transformation which preserves the sum over the channels.
The way I see it, in order to preserve the identity you'd need to consider individual feature planes, and not sum over them (something similar to grouped convolutions with number of groups = number of input channels).